### PR TITLE
feat(storage): use typescript

### PR DESCRIPTION
**Pull request by montezume**
#### Summary

Use typescript for the `storage` package.

### CODE REVIEW

**Comment by emmenko** [comment]
Hmm weird...maybe @tdeekens has an idea? Because I basically based the implementation like in flopflip...

**Comment by montezume** [comment]
I copied from [flopflip](https://github.com/tdeekens/flopflip/blob/master/packages/launchdarkly-adapter/tsconfig.json) and it seems to work now

Screen Shot 2019-06-07 at 1 59 21 PM

File: packages/storage/index.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export const put = (key: string, value: string) =>
+  window.localStorage.setItem(key, value);
```
Comment (montezume) [inline packages/storage/index.ts]: hey ma, I used static types 🙈

**Comment by montezume** [comment]
@tdeekens is on the way to the airport I think

**Comment by emmenko** [comment]
I just built it from your branch and I see this:

image

**Comment by montezume** [comment]
@emmenko yeah, isn't that the intended result?

File: packages/storage/tsconfig.json [hunk 2]
```diff
@@ -1,5 +1,4 @@
 {
   "extends": "../../tsconfig.json",
-  "files": ["./index.ts"],
-  "include": []
+  "files": ["./src/index.ts"]
```
Comment (montezume) [inline packages/storage/tsconfig.json]: did you try after removing include: []? that's the fix i got from flopflip.

**Comment by emmenko** [comment]
Yeah sorry, I didn't see that you updated your comment. I was referring to your question when you opened the PR.

Also, I pushed a commit to expose the version, apparently I missed that last time out.

File: packages/storage/tsconfig.json (hunk 2 above)
Comment (emmenko) [reply packages/storage/tsconfig.json]: Hmm, let me try again. I removed it because the sentry package also does not have it...

Comment (emmenko) [reply packages/storage/tsconfig.json]: Ooh ok, then we need to add it for each `tsconfig.json` I suppose. Good catch

Comment (emmenko) [reply packages/storage/tsconfig.json]: Wait, now I'm getting other errors 😱

Comment (emmenko) [reply packages/storage/tsconfig.json]: Ok this seems to fix it: https://github.com/commercetools/merchant-center-application-kit/pull/715/commits/19b7bb5500177e88c40040973190d86d649124f1

File: packages/sentry/tsconfig.json [hunk 3]
```diff
@@ -1,4 +1,5 @@
 {
   "extends": "../../tsconfig.json",
-  "files": ["./src/index.ts"]
+  "files": ["./src/index.ts"],
+  "include": ["./src/globals.d.ts"]
```
Comment (emmenko) [inline packages/sentry/tsconfig.json]: This is important in case we have custom declaration files

image

**Comment by montezume** [comment]
Thanks for the help @emmenko  😄
